src/SelfRoleManager.ts

```diff
@@ -43,48 +46,53 @@ export class SelfRoleManager extends EventEmitter {
   /**
    * Creates an instance of SelfRoleManager.
    * @param {Client} [client] The client that instantiated this Manager
-   * @param {SelfRoleOptions} [options={
+   * @param {SelfRoleOptions} [roleOptions={
    *     deleteAfterUnregistration: false,
    *     channelsMessagesFetchLimit: 3
    *   }]
    */
   constructor(
     client: Client,
-    options: SelfRoleOptions = {
+    roleOptions: SelfRoleOptions = {
       deleteAfterUnregistration: false,
       channelsMessagesFetchLimit: 3,
     }
   ) {
     super();
 
-    const intents = new Intents(client.options.intents);
-    if (!intents.has(Intents.FLAGS.GUILDS)) {
-      throw new Error('GUILDS intent is required to use this package!');
+    const intents = new IntentsBitField(client.options.intents);
+
+    if (!intents.has(IntentsBitField.Flags.Guilds)) {
+      throw new Error(
+        'Guilds intent bitfield is required to use this package!'
+      );
     }
-    if (!intents.has(Intents.FLAGS.GUILD_MEMBERS)) {
-      throw new Error('GUILD_MEMBERS intent is required to use this package!');
+    if (!intents.has(IntentsBitField.Flags.GuildMembers)) {
+      throw new Error(
+        'GuildMembers intent bitfield is required to use this package!'
```

Comment:
Updating the error message would be a huge breaking change and in this case is not essential to the update to discord.js v14. I am totally against introducing such changes in a context of a inner package version upgrade.

Reply:
I understand your point of view... I thought I did well as the intent fields changed, but no problem, I put the messages back